app/services/survey_import_service.rb

```diff
@@ -13,37 +15,62 @@ def import_item_name
   end
 
   def validity_errors
-    return ["Wrong format"] unless @file && Survey::SURVEY_FILE_TYPES.include?(@file.content_type)
+    return ["Wrong format"] unless @file && Survey::SURVEY_FILE_TYPES.include?(content_type)
 
     error_list = []
     error_list << "File empty" if csv.empty?
     error_list
   end
 
   def import!
-    ActiveRecord::Base.transaction do
-      # create a survey for the file
-      survey = create_survey
-      # create a survey question for each header
-      questions = create_survey_questions(survey, headers)
-      # create a submission for each row
-      create_submission_and_answers(survey, questions)
+    begin
+      ActiveRecord::Base.transaction do
+        # create a survey for the file
+        survey = create_survey
+        # create a survey question for each header
+        questions = create_survey_questions(survey, headers)
+        # create a submission for each row
+        create_submission_and_answers(survey, questions)
+      end
+    rescue ActiveRecord::RecordInvalid => e
+      @errors << e.message
     end
+
+    @notice = message_for_result
   end
 
   def create_survey
     survey = Survey.new(consultation: @consultation)
-    survey.original_file.attach(@file)
+    survey.original_file.attach(attachable)
     if survey.save!
       survey
     else
       @errors << survey.errors.messages
     end
   end
 
+  def attachable
+    @file.rewind
+    {
+      io: @file,
+      filename: filename,
+      content_type: content_type
+    }
+  end
+
+  def content_type
+    # Marcel is the content type detection library used by Active Storage
+    Marcel::MimeType.for @file, name: filename
```

Comment:
```suggestion
    Marcel::MimeType.for(@file, name: filename)
```

Just for consistency